readthedocs/redirects/models.py

```diff
@@ -143,8 +143,15 @@ def redirect_page(self, path, language=None, version_slug=None):
                 version_slug=version_slug)
             return to
 
-    def redirect_exact(self, path, **__):
-        if path == self.from_url:
+    def redirect_exact(self, path, language=None, version_slug=None):
+        if language and version_slug:
+            # reconstruct the full path for an exact redirect
+            full_path = '/{language}/{version_slug}/{path}'.format(
```

Comment:
Isn't it better here to use `self.get_full_path` which uses our `resolve_path` and handle more cases?